src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  return {
+    add(numb) {
+      result += numb;
+    },
+
+    subtract(numb) {
+      result -= numb;
+    },
+
+    multiply(numb) {
+      result *= numb;
+    },
+
+    divide(numb) {
+      result /= numb;
+    },
+
+    reset() {
+      result = 0;
+
+      return this;
+    },
+
+    operate(operation, number) {
+      operation(number);
```

Comment:
The `operate` method should verify that `operation` is a valid function before attempting to call it. Consider adding a check to ensure `operation` is a function.